manage.py

```diff
@@ -1,21 +1,153 @@
 #!/usr/bin/env python
 import os
+import re
+import socket
 import sys
+import time
+from typing import List
 
+import django
+import pyuwsgi
+from django.contrib.auth import get_user_model
+from django.core.management import call_command, execute_from_command_line
 
-# Separate definition to ease calling this in other scripts.
-def main():
+
+DEFAULT_ENVS = {
+    "DJANGO_SETTINGS_MODULE": "pydis_site.settings",
+    "SUPER_USERNAME": "admin",
+    "SUPER_PASSWORD": "admin",
+    "DEFAULT_BOT_API_KEY": "badbot13m0n8f570f942013fc818f234916ca531",
+}
+
+
+for key, value in DEFAULT_ENVS.items():
+    os.environ.setdefault(key, value)
+
+
+class SiteManager:
+    """
+    Manages the preparation and serving of the website.
+
+    Handles both development and production environments.
+
+    Usage:
+        manage.py run [option]...
+
+    Options:
+        --debug    Runs a development server with debug mode enabled.
+        --silent   Sets minimal console output.
+        --verbose  Sets verbose console output.
+    """
+
+    def __init__(self, args: List[str]):
+        self.debug = "--debug" in args
+        self.silent = "--silent" in args
+
+        if self.silent:
+            self.verbosity = 0
+        else:
+            self.verbosity = 2 if "--verbose" in args else 1
+
+        if self.verbosity and self.debug:
+            os.environ.setdefault("DEBUG", "true")
```

Comment:
Why would setting this also be dependent on verbosity? Does this environment variable affect verbosity?

Reply:
It's just ensuring that `--silent` isn't used when developing.

Reply:
Now that I'm trying to remember the exact thought process that made me write it, I actually can't recall the detailed reason. You're right, that part could probs just be removed. I'm going to do a big more digging in the Django docs just in case though.

Reply:
I think it might have been at a time when I was going to make debug the default. Adjusted now though anyhow, thanks!